tiledb/sm/query/sparse_unordered_with_dups_reader.cc

```diff
@@ -169,7 +169,7 @@ Status SparseUnorderedWithDupsReader<BitmapType>::dowork() {
   }
 
   // Load initial data, if not loaded already.
-  RETURN_NOT_OK(load_initial_data());
+  RETURN_NOT_OK(load_initial_data(subarray_.is_set()));
```

Comment:
Why is this conditional on the subarray having been set?

Reply:
If we have a subarray, dimensions will be loaded by default. This passes the information down to load_initial_data so it can now add the dimensions to the list of fields loaded by the query condition, if not loaded for range processing already. I added a small comment to clarify a little.

Just a note that for the sparse global order reader, coords are always loaded since they are required for the merge.